week-1-ts-ecosystem/w01e03-object-oriented-structure/src/feature/cart/components/cart.ts

```diff
@@ -0,0 +1,76 @@
+import { Product } from './product';
+
+interface Cart {
+  products: Product[];
+  cartType: 'Auction' | 'Buy Now' | 'Free';
+}
+
+class Cart {
+  constructor(cartType: 'Auction' | 'Buy Now' | 'Free') {
+    this.products = [];
+    this.cartType = cartType;
+  }
+
+  private isProductExist(id: number) {
+    return this.products.some((product) => product.id === id);
+  }
+
+  getAllProducts() {
+    return this.products;
+  }
+
+  addProduct(id: number, name: string, price: number, amount: number, type: 'Auction' | 'Buy Now' | 'Free') {
+    if (this.isProductExist(id)) {
+      return console.log(`You can not add products with this same ID`);
+    }
+    if (this.cartType === type) {
+      this.products.push(new Product(id, name, price, amount, type));
+    } else {
+      return console.log(`You can not add products with type ${type} to cart with type ${this.cartType}`);
+    }
+  }
+
+  getProductById(id: number) {
+    if (this.isProductExist(id)) {
+      return this.products.filter((product) => product.id === id);
+    } else {
+      return console.log(`There is no product with such ID.`);
```

Comment:
Jeśli twój `if` ma `return` to nie potrzebujesz `else {}` wystarczy tutaj poza `if` też dać `return`

Generalnei fajnie że obsługujesz wszyskie błędy - ale możesz iść "ostrzej" i rzucać tutaj `wyjątki` 
zamiast `console.log` robić:

```
throw new Error('There is no product with such ID.')
```